microsoft/testsuites/core/hv_module.py

```diff
@@ -0,0 +1,127 @@
+            "hid_hyperv",
+            "hv_balloon",
+            "hyperv_keyboard",
+        ]
+        uname = node.tools[Uname]
+        kernel_version = uname.get_linux_information().kernel_version_raw
+        config_path = f"/boot/config-{kernel_version}"
+
+        if node.execute(f"grep CONFIG_HYPERV_STORAGE=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_storvsc")
+        if node.execute(f"grep CONFIG_HYPERV_NET=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_netvsc")
+        if node.execute(f"grep CONFIG_HYPERV=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_vmbus")
+        if node.execute(f"grep CONFIG_HYPERV_UTILS=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_utils")
+        if node.execute(f"grep CONFIG_HID_HYPERV_MOUSE=y {config_path}").exit_code == 0:
+            hv_modules.remove("hid_hyperv")
+        if node.execute(f"grep CONFIG_HYPERV_BALLOON=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_balloon")
+        if node.execute(f"grep CONFIG_HYPERV_KEYBOARD=y {config_path}").exit_code == 0:
+            hv_modules.remove("hyperv_keyboard")
+
+        return hv_modules
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the list of given LIS kernel modules and verify if the version
+        matches with the Linux kernel release number.
+        """,
+        priority=0,
+    )
+    def verify_lis_modules_version(
+        self, case_name: str, log: Logger, node: RemoteNode
+    ) -> None:
+        if not (isinstance(node.os, CentOs) or isinstance(node.os, Redhat)):
+            log.info("Distro not supported")
+            raise SkippedException
+
+        hv_modules = self.get_modules(node)
+        rpm = node.tools[Rpm]
+        lis_installed = rpm.is_package_installed("microsoft-hyper-v")
+
+        modinfo = node.tools[Modinfo]
+        dmesg = node.tools[Dmesg]
+        lis_version = dmesg.get_lis_version()
+
+        if lis_installed:
+            for module in hv_modules:
+                module_version = modinfo.get_version(module)
+                assert_that(lis_version == module_version).described_as(
+                    f"Version of {module} is {module_version}, expected {lis_version}"
+                ).is_true()
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the presence of all LIS modules
+        """,
+        priority=0,
+    )
+    def lis_modules_check(self, case_name: str, log: Logger, node: RemoteNode) -> None:
+        hv_modules = self.get_modules(node)
+        distro_version = node.os.information.version
+
+        if isinstance(node.os, CentOs) or isinstance(node.os, Redhat):
+            rpm = node.tools[Rpm]
+            modprobe = node.tools[Modprobe]
+            lis_installed = rpm.is_package_installed("microsoft-hyper-v")
+
+            if distro_version >= "4.3.0" and lis_installed:
```

Comment:
Add comments to explain what's it used to check. We don't cover distro_version is lower than 4.3 now. The logic can be simpler.